pints/_log_likelihoods.py

```diff
@@ -237,6 +237,240 @@ def __call__(self, x):
+        = -\frac{n_tn_o}{2} \log 2 \pi
+        -\sum_{j=1}^{n_0}\sum_{i=1}^{n_t}\log \sigma _{\text{tot}, ij}
+        - \sum_{j=1}^{n_0}\sum_{i=1}^{n_t}
+        \frac{(X^{\text{obs}}_{ij} - f_j(t_i| \theta))^2}
+        {2\sigma ^2_{\text{tot}, ij}},
+
+    where :math:`n_o` is the number of outputs of the model,
+    :math:`X^{\text{obs}}_{ij}` is the observation at time point :math:`t_i`
+    of output :math:`j`, and
+    :math:`\sigma _{\text{tot}, ij}=\sigma _{\text{base}, j} +
+    \sigma _{\text{rel}, j}f_j(t_i| \theta)^{\eta _j}` is the total standard
+    deviation of the error at time :math:`t_i` of output :math:`j`.
+
+    Extends :class:`ProblemLogLikelihood`.
+
+    Parameters
+    ----------
+    ``problem``
+        A :class:`SingleOutputProblem` or :class:`MultiOutputProblem`. For a
+        single-output problem three parameters are added
+        (:math:`\sigma _{\text{base}}`, :math:`\eta`,
+        :math:`\sigma _{\text{rel}}`),
+        for a multi-output problem :math:`3n_o` parameters are added
+        (:math:`\sigma _{\text{base},1},\ldots , \sigma _{\text{base},n_o},
+        \eta _1,\ldots , \eta _{n_o}, \sigma _{\text{rel},1}, \ldots ,
+        \sigma _{\text{rel},n_o})`.
+    """
+
+    def __init__(self, problem):
+        super(CombinedGaussianLogLikelihood, self).__init__(problem)
+
+        # Get number of times and number of noise parameters
+        self._nt = len(self._times)
+        self._no = problem.n_outputs()
+        self._np = 3 * self._no
+
+        # Add parameters to problem
+        self._n_parameters = problem.n_parameters() + self._np
+
+        # Pre-calculate the constant part of the likelihood
+        self._logn = -0.5 * self._nt * self._no * np.log(2 * np.pi)
+
+    def __call__(self, parameters):
+        # Get parameters from input
+        noise_parameters = parameters[-self._np:]
+        sigma_base = np.asarray(noise_parameters[:self._no])
+        eta = np.asarray(noise_parameters[self._no:2 * self._no])
+        sigma_rel = np.asarray(noise_parameters[2 * self._no:])
+
+        # Evaluate noise-free model (n_times, n_outputs)
+        function_values = self._problem.evaluate(parameters[:-self._np])
+
+        # Compute error (n_times, n_outputs)
+        error = self._values - function_values
+
+        # Compute total standard deviation
+        sigma_tot = sigma_base + sigma_rel * function_values**eta
+
+        # Compute log-likelihood
+        # (inner sums over time points, outer sum over parameters)
+        log_likelihood = self._logn - np.sum(
+            np.sum(np.log(sigma_tot), axis=0)
+            + 0.5 * np.sum(error**2 / sigma_tot**2, axis=0))
+
+        return log_likelihood
+
+    def evaluateS1(self, parameters):
+        r"""
+        See :meth:`LogPDF.evaluateS1()`.
+
+        The partial derivatives of the log-likelihood w.r.t. the model
+        parameters are
+
+        .. math::
+            \frac{\partial \log L}{\partial \theta _k}
```

Comment:
I haven't been through these thoroughly, so would be good if someone else could check too. I am happy though that, as things currently stand, the tests show these derivs reproduce the special cases of the multiplicative model and GaussianLogLikelihood.